File: frontend/src/types/analytics.ts
export interface MonthlyRevenue {
  month: string
  revenue: number
}

export interface MonthlyCount {
  month: string
  count: number
}

export interface BillStatusCount {
  unpaid: number
  paid: number
  overdue: number
}

export interface ApiAnalyticsSummary {
  monthly_revenue: MonthlyRevenue[]
  monthly_tenants: MonthlyCount[]
  bill_status: BillStatusCount
  total_revenue: number
  avg_monthly: number
}

export interface ApiDashboardSummary {
  total_rooms: number
  occupied_rooms: number
  available_rooms: number
  maintenance_rooms: number
  occupancy_rate: number
  total_tenants: number
  active_contracts: number
  expiring_contracts: number
  unpaid_bills: number
  overdue_bills: number
  unpaid_amount: number
  overdue_amount: number
  revenue_this_month: number
}

File: backend/internal/domain/analytics.go
package domain

import "context"

type MonthlyRevenue struct {
	Month   string  `json:"month"`
	Revenue float64 `json:"revenue"`
}

type MonthlyCount struct {
	Month string `json:"month"`
	Count int    `json:"count"`
}

type BillStatusCount struct {
	Unpaid  int `json:"unpaid"`
	Paid    int `json:"paid"`
	Overdue int `json:"overdue"`
}

type AnalyticsSummary struct {
	MonthlyRevenue  []MonthlyRevenue `json:"monthly_revenue"`
	MonthlyTenants  []MonthlyCount   `json:"monthly_tenants"`
	BillStatus      BillStatusCount  `json:"bill_status"`
	TotalRevenue    float64          `json:"total_revenue"`
	AvgMonthly      float64          `json:"avg_monthly"`
}

type AnalyticsRepository interface {
	Summary(ctx context.Context, months int) (*AnalyticsSummary, error)
}
